libr/util/idpool.c

```diff
@@ -53,18 +72,34 @@ R_API bool r_id_pool_kick_id(RIDPool* pool, ut32 kick) {
 	}
 	if (kick == (pool->next_id - 1)) {
 		pool->next_id--;
+		if (pool->freed_ids) {
+			RBTreeIter iter = r_rbtree_last (pool->freed_ids);
+			while ((iter.len) && ((pool->next_id - 1) <= (ut32)(size_t)(iter.path[iter.len-1]->data))) {
+//				eprintf ("element in tree %d\n", (ut32)(size_t)(iter.path[iter.len-1]->data));
+				pool->next_id--;
+				r_rbtree_iter_prev (&iter);
+			}
+			pool->freed_ids->cmp = (RBTreeComparator)_del_cmp;
```

Comment:
I think there is no need to change the comparison function.

Reply:
well first i find out what the new next_id should be, then I delete all the ids that are stored in the tree, that are bigger or equal to that. I don't do the delition, in the same place, while figuring out the new next_id, bc I fear to break the iterator, when deleting in iterations